File: 50dias/cpp/32.cpp
#include <iostream>
#include <string>

using namespace std;

string LerTextoUsuario(const string& mensagem)
{
	cout << mensagem;
	string texto;
	getline(cin, texto);
	return texto;
}

string DecodificaTexto(const string& textoCodificado)
{
	string textoOriginal = "";
	for (auto letra : textoCodificado) {
		textoOriginal += letra - 3;
	}
	return textoOriginal;
}

int main()
{
	string textoCodificado = LerTextoUsuario("Digite um texto codificado: ");

	string textoOriginal = DecodificaTexto(textoCodificado);

	cout << "Texto decodificado: " << textoOriginal << endl;
	return 0;
}

File: 50dias/python/32.py

def LerTextoUsuario(mensagem):
    texto = input(mensagem)
    return texto

def DecodificaTexto(textoCodificado):
    textoOriginal = ""
    for letra in textoCodificado:
        textoOriginal += chr(ord(letra) - 3)
    return textoOriginal

def main():
    textoCodificado = LerTextoUsuario("Digite um textoCodificado: ")
    textoDecodificado = DecodificaTexto(textoCodificado)

    print('Texto decodificado: ' + textoDecodificado)

main()